sphereon-kmp-crypto-kms-azure/src/jvmMain/kotlin/com/sphereon/crypto/kms/azure/AzureKeyVaultMapper.jvm.kt

```diff
@@ -0,0 +1,182 @@
+}
+
+fun ByteArray.toBase64UrlString(): String =
+    java.util.Base64.getUrlEncoder().withoutPadding().encodeToString(this)
+
+fun KeyOperation.toKeyOperations(): KeyOperations {
+    return when (this) {
+        KeyOperation.ENCRYPT -> KeyOperations.ENCRYPT
+        KeyOperation.DECRYPT -> KeyOperations.DECRYPT
+        KeyOperation.SIGN -> KeyOperations.SIGN
+        KeyOperation.VERIFY -> KeyOperations.VERIFY
+        KeyOperation.WRAP_KEY -> KeyOperations.WRAP_KEY
+        KeyOperation.UNWRAP_KEY -> KeyOperations.UNWRAP_KEY
+        else -> {
+            throw SignClientException("Unsupported key operation for Azure Key Vault")
+        }
+    }
+}
+
+fun KeyVaultKey.toJwk(): Jwk {
+    val jsonWebKey: JsonWebKey = this.key
+
+    return Jwk.Builder()
+        .withKid(jsonWebKey.id) // Key ID
+        .withKty(jsonWebKey.keyType?.toString()?.let { JwaKeyType.Static.fromValue(it) }) // Key type
+        .withAlg(mapJwkToAlgorithm(jsonWebKey)) // Algorithm
+        .withN(jsonWebKey.n?.toBase64UrlString())
+        .withE(jsonWebKey.e?.toBase64UrlString())
+        .withX(jsonWebKey.x?.toBase64UrlString())
+        .withY(jsonWebKey.y?.toBase64UrlString())
+        .withKeyOps(jsonWebKey.keyOps?.map { JoseKeyOperations.Static.fromValue(it.toKeyOperations().jose.value) }
+            ?.toTypedArray())
+        .build()
+}
+
+fun KeyOperations.toAzureKeyOperation(): KeyOperation {
+    return when (this) {
+        KeyOperations.ENCRYPT -> KeyOperation.ENCRYPT
+        KeyOperations.DECRYPT -> KeyOperation.DECRYPT
+        KeyOperations.SIGN -> KeyOperation.SIGN
+        KeyOperations.UNWRAP_KEY -> KeyOperation.UNWRAP_KEY
+        KeyOperations.VERIFY -> KeyOperation.VERIFY
+        KeyOperations.WRAP_KEY -> KeyOperation.WRAP_KEY
+        KeyOperations.DERIVE_BITS -> throw SignClientException("Azure Key Vault does not support DERIVE_BITS operation")
+        KeyOperations.DERIVE_KEY -> throw SignClientException("Azure Key Vault does not support DERIVE_KEY operation")
+        KeyOperations.MAC_CREATE -> throw SignClientException("Azure Key Vault does not support MAC_CREATE operation")
+        KeyOperations.MAC_VERIFY -> throw SignClientException("Azure Key Vault does not support MAC_VERIFY operation")
+    }
+}
+
+fun mapJwkToAlgorithm(jwk: JsonWebKey): JwaAlgorithm? {
+    return when (jwk.keyType) {
+        KeyType.EC -> when {
+            jwk.curveName.toString() == "P-256" -> JwaAlgorithm.ES256
+            jwk.curveName.toString() == "P-384" -> JwaAlgorithm.ES384
+            jwk.curveName.toString() == "P-521" -> JwaAlgorithm.ES512
+            else -> null
+        }
+        KeyType.RSA -> JwaAlgorithm.RS256
+        else -> null
+    }
+}
+
+fun Curve.toAzureKeyCurveName(): KeyCurveName {
+    return when (this) {
+        Curve.P_256 -> KeyCurveName.P_256
+        Curve.P_384 -> KeyCurveName.P_384
+        Curve.P_521 -> KeyCurveName.P_521
+        Curve.Ed25519 -> throw SignClientException("Curve Ed25519 is not supported")
```

Comment:
else throw this.value is not supported. Then no matter what gets added to the enum in the future it already is exhaustive